pkg/volume/quobyte/quobyte.go

```diff
@@ -158,6 +158,23 @@ func (plugin *quobytePlugin) newMounterInternal(spec *volume.Spec, pod *api.Pod,
 		return nil, err
 	}
 
+	if pod != nil {
+		// We use the service account if it's not the default service account
+		if len(pod.Spec.ServiceAccountName) != 0 && pod.Spec.ServiceAccountName != "default" {
+			source.User = pod.Spec.ServiceAccountName
+			source.Group = pod.Spec.ServiceAccountName
```

Comment:
> As Kubernetes user info I understand the "user" and "group" that are used to create the resource via the API (like with a static token file) and all "children" resources (pods) would inherit the user and group from the parent resource like a RC.

That is not how the system works today... if a user creates a RC, the system becomes responsible for fulfilling that RC's spec. If the user is removed from the system, but the RC remains, it should still be valid and pods should still be able to be created from it.

> These are just names that are used to map the access to the volume to, e.q. any access to a volume get mapped to the specified user and group (which are in the most cases unix user/groups).

Service account name is not connected in any way to fs users and groups today. I'd expect more discussion before committing to making that connection here. @pmorie, any feedback on how other volume plugins select owning uids/gids?